exca/test_utils.py

```diff
@@ -183,6 +183,49 @@ def test_optional_discriminator(caplog: tp.Any) -> None:
     assert out == expected
 
 
+class RecursiveLeaf(pydantic.BaseModel):
+    model_config = pydantic.ConfigDict(extra="forbid")
+    edge_type: tp.Literal["leaf"] = "leaf"
+
+
+class RecursiveEdge(pydantic.BaseModel):
+    model_config = pydantic.ConfigDict(extra="forbid")
+    infra: exca.TaskInfra = exca.TaskInfra(cluster=None, folder="/tmp/")
```

Comment:
please remove the folder, and use a tmp_path: Path parameter in the test instead (i'd rather avoid border effects in tests)